### PR TITLE
Edge array Python API

### PR DESCRIPTION
**Pull request by GertjanBisschop**
First pass at low-level Python and Python API for `Tree.edge_array`.

### CODE REVIEW

File: python/tests/test_highlevel.py [hunk 1]
```diff
@@ -2364,7 +2364,9 @@ def modify(ts, func):
     def test_tree_node_edges(self):
         for ts in get_example_tree_sequences():
             edge_visited = np.zeros(ts.num_edges, dtype=bool)
-            for mapping, tree in zip(ts._tree_node_edges(), ts.trees()):
+            # for mapping, tree in zip(ts._tree_node_edges(), ts.trees()):
```
Comment (jeromekelleher) [inline python/tests/test_highlevel.py:2367]: I guess we can delete the ``_tree_node_edges`` thing now as well (not sure where that's come from?)

**Review by jeromekelleher — COMMENTED** [review]
LGTM!

File: python/CHANGELOG.rst [hunk 2]
```diff
@@ -1,3 +1,13 @@
+--------------------
+[0.5.1] - 2022-XX-XX
+--------------------
+
+**Changes**
+
+- Add ``Tree.edge_array`` and ``Tree.edge``. Returns the edge id of the encoding
+  the relationship of each node with its parent.
+  (:user:`GertjanBisschop`)
```
Comment (benjeffery) [inline python/CHANGELOG.rst:9]: Is there an issue and PR num we can put here?

**Review by benjeffery — APPROVED** [review]
LGTM, one question about the change log.

File: python/CHANGELOG.rst (hunk 2 above)
Comment (GertjanBisschop) [reply python/CHANGELOG.rst:9]: Good point. Will make an issue and add a PR num. And add the changes to the 0.5.0 release section.

File: python/tests/test_highlevel.py (hunk 1 above)
Comment (jeromekelleher) [reply python/tests/test_highlevel.py:2367]: In general we don't want to keep commented out bits lying around -that's what git is for!

**Comment by GertjanBisschop** [comment]
Resolves #2361

**Comment by benjeffery** [comment]
@GertjanBisschop I'll fix up these comments and merge, as this is the last PR before 0.5.0!

**Comment by benjeffery** [comment]
CircleCI still AWOL.

**Comment by benjeffery** [comment]
@mergifyio refresh

**Pull request merged**